blog/content/post/2021-06-14-fwupd-freebsd-status.md

```diff
@@ -0,0 +1,1497 @@
+---
+title: fwupd on FreeBSD - Status Update
+abstract: 'An update on the current status of the BSD port of fwupd.'
+cover: /covers/fwupd_bsd.png
+author: michal.kopec
+layout: post
+published: true
+date: 2021-06-14
+archives: "2021"
+
+tags:
+  - fwupd-for-bsd
+  - bsd
+  - fwupd
+categories:
+  - Firmware
+
+---
+
+This is the third entry in the series documenting porting fwupd to *BSD
+distributions. If you haven't read the
+[previous](https://blog.3mdeb.com/2021/2021-02-16-fwupd-compilation-under-freebsd/)
+[entries](https://blog.3mdeb.com/2021/2021-02-16-fwupd-compilation-under-freebsd/),
+I encourage you to do so.
+
+---
+
+## Introduction
+
+In this blog post, I will document how we created a working port of fwupd for
+FreeBSD as well as the biggest challenges we encountered.
+
+We found we had to implement 3 basic functionalities in order to have a basic
+FreeBSD port:
+
+- Gathering information about applicable updates for attached devices,
+- Updating firmware of devices attached via USB,
+- Updating the system firmware via the UEFI Update Capsule mechanism.
+
+## Identifying available updates
+
+One difference between the Linux and FreeBSD APIs is the lack of `memfd_create(),`
```

Comment:
Not true, it was added in FreeBSD 13.0, but we started implementation for version 12.2, where it is not included.